plugin/pkg/shoot/managedseed/admission_test.go

```
@@ -0,0 +1,207 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package managedseed_test
+
+import (
+	"context"
+	"errors"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+	corev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	seedmanagementv1alpha1 "github.com/gardener/gardener/pkg/apis/seedmanagement/v1alpha1"
+	corefake "github.com/gardener/gardener/pkg/client/core/clientset/internalversion/fake"
+	seedmanagementfake "github.com/gardener/gardener/pkg/client/seedmanagement/clientset/versioned/fake"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	. "github.com/gardener/gardener/plugin/pkg/shoot/managedseed"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/client-go/testing"
+)
+
+const (
+	name      = "foo"
+	namespace = "garden"
+)
+
+var _ = Describe("ManagedSeed", func() {
+	Describe("#Validate", func() {
+		var (
+			shoot                *core.Shoot
+			managedSeed          *seedmanagementv1alpha1.ManagedSeed
+			coreClient           *corefake.Clientset
+			seedManagementClient *seedmanagementfake.Clientset
+			admissionHandler     *ManagedSeed
+		)
+
+		BeforeEach(func() {
+			shoot = &core.Shoot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+			}
+
+			managedSeed = &seedmanagementv1alpha1.ManagedSeed{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+				Spec: seedmanagementv1alpha1.ManagedSeedSpec{
+					Shoot: seedmanagementv1alpha1.Shoot{
+						Name: name,
+					},
+				},
+			}
+
+			admissionHandler, _ = New()
```

Please don't ignore errors in tests

---

OK

---

Done